internal/cli/atlas/customdbroles/update.go

```diff
@@ -0,0 +1,123 @@
+// limitations under the License.
+
+package customdbroles
+
+import (
+	"errors"
+
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/convert"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const updateTemplate = "Custom Database Role successfully updated.\n"
+
+type UpdateOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	roleName       string
+	action         []string
+	inheritedRoles []string
+	append         bool
+	store          store.DatabaseRoleUpdater
+}
+
+func (opts *UpdateOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(config.Default())
+	return err
+}
+
+func (opts *UpdateOpts) Run() error {
+	var role *atlas.CustomDBRole
+	var err error
+	if opts.append {
+		if role, err = opts.store.DatabaseRole(opts.ConfigProjectID(), opts.roleName); err != nil {
+			return err
+		}
+	}
+
+	out, err := opts.store.UpdateDatabaseRole(opts.ConfigProjectID(), opts.roleName, opts.newCustomDBRole(role))
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(out)
+}
+
+func (opts *UpdateOpts) newCustomDBRole(role *atlas.CustomDBRole) *atlas.CustomDBRole {
+	out := &atlas.CustomDBRole{
+		Actions:        []atlas.Action{},
+		InheritedRoles: []atlas.InheritedRole{},
+	}
+
+	if len(opts.action) > 0 {
+		out.Actions = convert.BuildAtlasActions(opts.action)
+	}
+
+	if len(opts.inheritedRoles) > 0 {
+		out.InheritedRoles = convert.BuildAtlasInheritedRoles(opts.inheritedRoles)
+	}
+
+	if opts.append {
+		out.Actions = append(out.Actions, role.Actions...)
+		out.InheritedRoles = append(out.InheritedRoles, role.InheritedRoles...)
+	}
+
+	return out
+}
+
+func (opts *UpdateOpts) validate() error {
+	if len(opts.action) == 0 && opts.inheritedRoles == nil {
+		return errors.New("you must provide either actions or inherited roles")
+	}
+	return nil
+}
+
+// mongocli atlas dbrole(s) update create <roleName> --privilege action[@dbName.collection] --inheritedRole role@db --append
+func UpdateBuilder() *cobra.Command {
+	opts := &UpdateOpts{}
+	cmd := &cobra.Command{
+		Use:   "update",
+		Short: updateDBRole,
+		Args:  cobra.ExactArgs(1),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			return opts.PreRunE(
+				opts.ValidateProjectID,
+				opts.initStore,
+				opts.InitOutput(cmd.OutOrStdout(), updateTemplate),
+				opts.validate,
+			)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.roleName = args[0]
+			return opts.Run()
+		},
+	}
+
+	cmd.Flags().StringSliceVar(&opts.inheritedRoles, flag.InheritedRole, []string{}, usage.InheritedRoles)
+	cmd.Flags().StringSliceVar(&opts.action, flag.Privilege, []string{}, usage.PrivilegeAction)
+	cmd.Flags().BoolVar(&opts.append, flag.Append, false, usage.Append)
```

Comment:
[not-blocking] I'm not sure about this flag, I know what you want but this may be confusing

Reply:
I like the idea of having a way to append roles & actions to an existing one. I tested it with e2e test and manually and it works pretty well

Reply:
yes, I like the idea, and I'm pretty sure it works I'm just worry of the implementation for users, ie, having to remember to use `--append` if they want to add instead of replace it's the part I worry could be confusing for some.

I'd say we could have two extra commands, `add actions` and `add roles` that have this particular behaviour and are more clear on intent than a flag but we should discuss further

Reply:
could I merge this PR if this is the only thing blocking it? I would create a new ticket to update this command if we decided in the Spec that we want to address this in another way.

Reply:
Let's remove the flag for know, or mark it as hidden if you want and we can review latter making it official

Reply:
done